### PR TITLE
GPS: Don't auto detect nmea instances

### PR DESCRIPTION
**Pull request by WickedShell**
As discussed on the dev call this removes auto detecting NMEA instances, as it's almost always an error, and it's also the worst driver in terms of information we can reliably expect to decode with it. This also has the slight bonus that if you are using a NMEA GPS the detection time is faster, as we no longer have to wait for a complete cycle of the possible baud rates.

Second commit leveraged the fact that we no longer need a timestamp for NMEA detection reasons and transformed the detect started time stamping to instead just be a bool flag. I looked briefly at fixing the SBF/NOVA/GSOF incorrect baud rate prints, but I didn't come across a simple way to do implement that part in the same pass, so I've left that for future work.

### CODE REVIEW

File: libraries/AP_GPS/AP_GPS.h [hunk 1]
```diff
@@ -419,9 +419,9 @@ class AP_GPS
 
     // state of auto-detection process, per instance
     struct detect_state {
-        uint32_t detect_started_ms;
         uint32_t last_baud_change_ms;
         uint8_t current_baud;
+        bool auto_detect_baud;
```
Comment (OXINARF) [inline libraries/AP_GPS/AP_GPS.h]: auto_detect_baud -> auto_detect**ed**_baud or change everywhere else :slightly_smiling_face:

Comment (WickedShell) [inline libraries/AP_GPS/AP_GPS.h]: You know... That's how I was reading it in my head, and never actually noticed :P

**Comment by WickedShell** [comment]
Pushed the variable name fix. I had debated the printing the auto detect part up earlier. The advantage of not doing so is that I don't think any of those GPS's respond at 4800 anyways, but it's more consistent if it is. (So I did move it).

Regarding PR name... Yeah I had an actual/normal one but then saw something I wanted to change, pushed the changes then just refreshed the page and forgot to fix the title :)

**Comment by OXINARF** [comment]
Merged, thanks!

**Pull request merged**